src/docker.rs

```diff
@@ -183,15 +185,42 @@ pub fn run(target: &Target,
     }
 
     docker
-        .arg(&image(toml, target)?)
+        .arg(&image(toml, target, verbose)?)
         .args(&["sh", "-c", &format!("PATH=$PATH:/rust/bin {:?}", cmd)])
         .run_and_get_status(verbose)
 }
 
-pub fn image(toml: Option<&Toml>, target: &Target) -> Result<String> {
+fn build_docker_image(toml: &Toml, target: &Target, verbose: bool) -> Result<String> {
+    let mut image_id_filename = std::env::temp_dir();
+    image_id_filename.push(format!("{}.iid", std::process::id()));
+
+    let mut docker = docker_command("build")?;
+    docker
+        .arg(toml.context(target)?.map(|s| s.to_owned()).unwrap())
+        .args(&["--iidfile", &image_id_filename.clone().into_os_string().into_string().unwrap()]);
+    if let Some(dockerfile) = toml.dockerfile(target)? {
+        docker.args(&["-f", dockerfile]);
+    }
+    if verbose {
+        docker.run(verbose)?;
+    } else {
+        docker.run_and_get_stdout(verbose)?;
+    }
+    let mut image_id = std::fs::read_to_string(&image_id_filename)?;
+    image_id.retain(|c| c != '\n');
+    std::fs::remove_file(&image_id_filename).unwrap_or_else(
```

Comment:
I kind-of want to do this within a ScopeGuard, or import and use https://docs.rs/tempfile/3.1.0/tempfile/struct.NamedTempFile.html . What is `cross`'s policy on adding new dependencies?